beluga-mqtt/tests/connection.rs

```diff
@@ -0,0 +1,48 @@
+use core::panic;
```

Comment:
I would like to see a "reconnect" test as well where we:
1. connect
2. intentionally restart the broker
3. check if the clients reconnected